src/index.ts

```diff
@@ -156,99 +242,93 @@ export const findNextItem = (stimuli: Stimulus[], theta = 0, method = 'MFI', dee
+    }
 
-  /**
-   * find the item in a given array that has the difficulty closest to the target value
-   * @param arr Array<Stimulus> - an array of stimulus
-   * @param target number - ability estimate
-   * @returns {number} the index of arr
-   */
-  function findClosest(arr: Array<Stimulus>, target: number) {
-    const n = arr.length;
-    // Corner cases
-    if (target <= arr[0].difficulty) return 0;
-    if (target >= arr[n - 1].difficulty) return n - 1;
-    // Doing binary search
-    let i = 0,
-      j = n,
-      mid = 0;
-    while (i < j) {
-      mid = Math.ceil((i + j) / 2);
-      if (arr[mid].difficulty == target) return mid;
-      // If target is less than array
-      // element,then search in left
-      if (target < arr[mid].difficulty) {
-        // If target is greater than previous
-        // to mid, return closest of two
-        if (mid > 0 && target > arr[mid - 1].difficulty) return getClosest(arr, mid - 1, mid, target);
-        // Repeat for left half
-        j = mid;
-      }
-      // If target is greater than mid
-      else {
-        if (mid < n - 1 && target < arr[mid + 1].difficulty) return getClosest(arr, mid, mid + 1, target);
-        i = mid + 1; // update i
+    /**
+     * find the item in a given array that has the difficulty closest to the target value
+     * @param arr Array<Stimulus> - an array of stimulus
+     * @param target number - ability estimate
+     * @returns {number} the index of arr
+     */
+    private static findClosest(arr: Array<Stimulus>, target: number) {
+      const n = arr.length;
+      // Corner cases
+      if (target <= arr[0].difficulty) return 0;
+      if (target >= arr[n - 1].difficulty) return n - 1;
+      // Doing binary search
+      let i = 0,
+          j = n,
+          mid = 0;
+      while (i < j) {
+        mid = Math.ceil((i + j) / 2);
+        if (arr[mid].difficulty == target) return mid;
+        // If target is less than array
+        // element,then search in left
+        if (target < arr[mid].difficulty) {
+          // If target is greater than previous
+          // to mid, return closest of two
+          if (mid > 0 && target > arr[mid - 1].difficulty) return Cat.getClosest(arr, mid - 1, mid, target);
+          // Repeat for left half
+          j = mid;
+        }
+        // If target is greater than mid
+        else {
+          if (mid < n - 1 && target < arr[mid + 1].difficulty) return Cat.getClosest(arr, mid, mid + 1, target);
+          i = mid + 1; // update i
+        }
       }
+      // Only single element left after sear
+      return mid;
     }
-    // Only single element left after search
-    return mid;
 
-    function getClosest(arr: Array<Stimulus>, val1: number, val2: number, target: number) {
-      if (target - arr[val1].difficulty >= arr[val2].difficulty - target) return val2;
-      else return val1;
+    private static getClosest(arr: Array<Stimulus>, val1: number, val2: number, target: number) {
```

Comment:
Same as above. Consider extracting this from the `Cat` class and writing as a standalone function in `src/utils.ts`.